usage/usageplot.py

```diff
@@ -85,23 +93,107 @@ def make_plot(title, times, nodes, name):
+    """
     plt.figure(figsize=(8, 8))
     plt.plot(times, nodes, 'b', drawstyle="steps-post")
     plt.ylim(0, 50)
     plt.xlim(xmin=0)
     plt.grid(linestyle=':')
-    plt.fill_between(times, 0, nodes, step="post", facecolor='b', alpha=0.25)
+    
+    if color:
+        start_end = get_first_last(jobs)
+        colors = {'singleFrame': 'c', 'mosaic': 'xkcd:yellow', 'coadd': 'g',
+                  'multiband': 'b', 'unknown': 'r'}
+        hatch = {'singleFrame': '///', 'mosaic': '\\\\', 'coadd': '...',
+                 'multiband': 'x', 'unknown': '**'}
+
+        for key in start_end.keys():
+            for val_tup in start_end[key]:
+                plt.fill_between(times[val_tup[0]:val_tup[1]+2], 0,
+                                 nodes[val_tup[0]:val_tup[1]+2], hatch=hatch[key],
+                                 step="post", facecolor=colors[key], alpha=0.5,
+                                 label=key if key not in
+                                 plt.gca().get_legend_handles_labels()[1] else '')
+        plt.legend(title='Code Name')
+    
+    else:
+        plt.fill_between(times, 0, nodes, step="post", facecolor='b', alpha=0.25)
+
     plt.xlabel(r'time [h]', fontsize=16)
     plt.ylabel(r'$\ N_{node} $', fontsize=16)
     plt.tick_params(axis='both', which='major', labelsize=14)
     plt.title(title, fontsize=18)
     plt.savefig(name + ".png")
 
 
+def get_first_last(jobs):
+    """Gets the first and the last index of the code names of the same type,
+    (i.e.: will return the first and last index of the singleFrameDriver jobs)
+    which will allow for color-coding in make-plot.
+
+    Parameters
+    ----------
+    jobs : `list` of `str`
+        list of the job names from SLURM at a given time step
+
+     Returns
+    -------
+    start_end : `dict` of `list` of `tuple`
+        dictionary containing code names and the start and stop indexes
+        of the code runs.
+    """
+    names = dict()
+    # Creates dictionary of the first two letters of a job name & the indexes
+    # where those job names can be found.
+    for k in range(len(jobs)):
+        for j in jobs[k]:
+            if j[:2] in names:
+                names[j[:2]].append(k)
+            else:
+                names[j[:2]] = [k]
+
+    data_final = dict()
+    # Translates common entries in the names dictionary to the actual code
+    # names. If the key in names doesn't match with any of those below, the
+    # code will be marked as "unknown"
+    for key in names.keys():
+        if key == 'Wi':
+            data_final['singleFrame'] = names[key]
+        elif key == 'Co':
+            data_final['singleFrame'] += names[key]
+        elif 'mo' == key:
+            data_final['mosaic'] = names[key]
+        elif 'co' == key:
+            data_final['coadd'] = names[key]
+        elif 'mt' == key:
+            data_final['multiband'] = names[key]
+        else:
+            data_final['unknown'] = names[key]
+
```

Comment:
1. While line 164 is completely fine, it is common to omit `keys()` method -- iterating over a dictionary implicitly iterates over its keys.

2. Your code implicitly assumes that the key `Wi` comes before `Co`. Otherwise it will try to append a list to yet undefined value. Key ordering in a dictionary is rather broad and complex (see [this](https://stackoverflow.com/a/15479974) Stackoverflow answer for example). As a rule of thumb, I would make no assumptions about the order they will be returned.